### PR TITLE
Use view hooks for export index view

### PR DESCRIPTION
**Pull request by aapomm**
Instead of using the `@plugins` variable and iterating through every export plugins, we are adding a render view hook that export plugins should use. 

One important change here is adding an argument to `render_view_hooks` to specify the addon feature:

```
def render_view_hooks(partial, feature = :addon, locals: {})
```

to include the export plugins.

### CODE REVIEW

File: app/views/export/index.html.erb [hunk 1]
```diff
@@ -1,40 +1,26 @@
...
+      <p class="mb-0">Check the Export section of your <code><%= Rails.root.join('Gemfile.plugins') %></code> file.</p>
     </div>
-  <% else %>
+  </div>
+<% else %>
 
-    <ul id="plugin-chooser" class="tabs-container nav nav-tabs">
-      <% @plugin_info.each do |info| %>
+  <ul id="plugin-chooser" class="tabs-container nav nav-tabs">
+    <%= render_view_hooks 'export/plugin_description', :export%>
```
Comment (sean-yeoh) [inline app/views/export/index.html.erb:13]:
```suggestion
    <%= render_view_hooks 'export/plugin_description', :export% >
```

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -16,8 +16,8 @@ def markup(text)
     result[:output].to_s.html_safe
   end
 
-  def render_view_hooks(partial, locals: {})
-    Dradis::Plugins::with_feature(:addon).sort_by(&:plugin_description).each do |plugin|
+  def render_view_hooks(partial, feature = :addon, locals: {})
```
Comment (sean-yeoh) [inline app/helpers/application_helper.rb:19]: @aapomm We've added this functionality in Pro during the permissions cycle https://github.com/secroots/dradis-pro/blob/develop/app/helpers/application_helper.rb#L23.

Can you take care of the merge conflicts once this is merged?

Comment (aapomm) [reply app/helpers/application_helper.rb]: I missed that one. Thanks!

**Comment by etdsoft** [comment]
@aapomm why do we need the `:feature` change? 

Traditionally we've relied on plugins resitering the `:addon` if they'd like to provide a View extension (no matter where), and then on the implementation of `.render_view_hooks()` to rescue from `ActionView::MissingTemplate` if the given plugin didn't provide a particular view for a particular hook.

I'm worried about the proliferation of approaches around `render_view_hooks`, for instance: should we use `feature: :calculator` when building the Issues#show tab? Or `feature: :menu` when rendering the Mintcreek Add-ons menu?

I'm not sure what the best way forward here is.

File: app/views/export/index.html.erb [hunk 3]
```diff
@@ -1,40 +1,26 @@
...
+      <p class="mb-0">Check the Export section of your <code><%= Rails.root.join('Gemfile.plugins') %></code> file.</p>
     </div>
-  <% else %>
+  </div>
+<% else %>
 
-    <ul id="plugin-chooser" class="tabs-container nav nav-tabs">
-      <% @plugin_info.each do |info| %>
+  <ul id="plugin-chooser" class="tabs-container nav nav-tabs">
+    <%= render_view_hooks 'export/plugin_description', feature: :export %>
```
Comment (etdsoft) [inline app/views/export/index.html.erb:13]: I'd like for us to keep naming consistent: for Issues#show calculator tabs we're using:

```erb
<%= render_view_hooks 'issues/show-tabs' %>
```

and 

```
<%= render_view_hooks 'issues/show-content' %>
```

What about using `export/index-tabs` and `export/index-content` here?

**Pull request merged**